### PR TITLE
test(quick-install): Enable quick and experimential installation in CI

### LINKED ISSUE #10616: Ensure that quick-install YAML is run in CI
A recent breakage on master for quick-install YAML was not detected until developers manually ran the quick-install YAML. One aspect is that developers can neglect to update the quick-install YAML. #10425 follows up on fixing that.

The second aspect is that we should automate deployment of the quick-install YAML from the repo during CI for basic validation that it works.

I briefly grepped for `quick-install` in the repo and couldn't find any sign of such a test.

Related: #10425

### PR DESCRIPTION
**Pull request by sayboras**
Fixes: #10616

```release-note
Enable quick and experimential installation in CI
```

### CODE REVIEW

**Review by aanm — APPROVED** [review]

**Review by aanm — COMMENTED** [review]
@sayboras non blocking but would it also be possible to test it in a v6 only environment in a parallel GH action as well?

**Comment by sayboras** [comment]
@aanm yes, i saw a few issues recently with ipv6, so just thought about adding related test as well. Will do it tomorrow in separate PR if you don’t mind.

**Comment by aanm** [comment]
> @aanm yes, i saw a few issues recently with ipv6, so just thought about adding related test as well. Will do it tomorrow in separate PR if you don’t mind.

great! thank you!

**Comment by Rolinh** [comment]
@aanm @sayboras I've been trying to have an IPv6 only cluster running with `kind` lately and ran into multiple issues (probably some that you saw). Some of the required fixes are already merged into `master` but it's not sufficient yet to have an IPv6 only cluster working with `kind`. You'll likely run into more issues when working on this @sayboras so feel free to ping me if you need help.

**Comment by pchaigno** [comment]
I don't think GitHub Actions support IPv6 anyway: https://github.community/t/github-actions-ipv6-network-issues/18004.

File: install/kubernetes/Makefile [hunk 1]
```diff
@@ -53,7 +53,10 @@ update-versions:
 	@# Fix up the managed etcd version, as that has its own scheme
 	$(QUIET)sed -i 's/'$(VERSION)'/'$(MANAGED_ETCD_VERSION)'/' $(MANAGED_ETCD_PATH)
 
+lint:
+	$(QUIET)helm lint --with-subcharts --values ./cilium/values.yaml ./cilium
+
 clean:
 	$(RM) $(QUICK_INSTALL) $(EXPERIMENTAL_INSTALL)
 
-.phony: all clean update-versions
```
Comment (joestringer) [inline install/kubernetes/Makefile:59]: Should we also add `quick-install` / `experimental-install` to the `.PHONY` target?

**Comment by sayboras** [comment]
> I don't think GitHub Actions support IPv6 anyway: https://github.community/t/github-actions-ipv6-network-issues/18004.

Oh right, I am using local machine as _self-hosted_ github action for debugging purpose before sending PR, and my ubuntu is having IPv6 enabled. Maybe that is why I have impression that github action supports ipv6 :cry:

File: install/kubernetes/Makefile (hunk 1 above)
Comment (sayboras) [reply install/kubernetes/Makefile:59]: yes, it's my bad. Thanks :100:

**Review by christarazi — APPROVED** [review]
Awesome, thanks for the PR! :tada:

**Review by michi-covalent — APPROVED** [review]

**Review by vadorovsky — APPROVED** [review]

**Comment by sayboras** [comment]
hi, just want to highlight post-merge action for this PR. As I have done some renaming on github action job, we might need to change required check in github accordingly. Currently, all the builds will just have `quick-install` status waiting. Thanks.

**Comment by pchaigno** [comment]
> hi, just want to highlight post-merge action for this PR. As I have done some renaming on github action job, we might need to change required check in github accordingly. Currently, all the builds will just have `quick-install` status waiting. Thanks.

/cc @aanm

**Pull request merged**